flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -679,4 +708,19 @@ private void startScheduling(final SchedulerNG scheduler) {
 		scheduler.setMainThreadExecutor(ComponentMainThreadExecutorServiceAdapter.forMainThread());
 		scheduler.startScheduling();
 	}
+
+	private static List<ExecutionVertexID> getAllExecutionVertexIdsSorted(final JobGraph jobGraph) {
```

Comment:
I'd rather create a copy of `deployedExecutionVertices`, sort it using your comparator, and compare for list equality. The comparator is already unit tested so it's result can be trusted. This way we would avoid re-implementing logic from the comparator.

Reply:
A copy of `deployedExecutionVertices` does not help to verify whether comparator is invoked  and whether the ordered result is respected in `DefaultScheduler`.
Agreed that we can trust the result of the comparator so we did not test each possible case but just verifies whether the `deployedExecutionVertices` is really ordered in the most common case.
`getAllExecutionVertexIdsSorted` is created for code reusability, so that we do not need to iterate each vertex to generate `ExecutionVertexID`s in a certain case.

Reply:
not the case anymore